### DISCUSSION TITLE
Scene/Avatar thumbnail failed to load

### DISCUSSION THREAD

**Discussion by hungleon2112** [mozilla/hubs#4831] [Hubs Cloud Questions] [answered]
Hello! 
After import all the scene and avatar, when I tried to create a room and setup some options like chose scene and avatar, the thumbnail of them do not load correctly.

![image](https://user-images.githubusercontent.com/10544674/141239972-50fc4cd3-9db6-47e2-825f-243b1bbb2577.png)
![image](https://user-images.githubusercontent.com/10544674/141239990-fc4a0e1c-17f3-42ef-9642-ce256988fa11.png)

I've checked in Developer Tools and I see the error 502
![image](https://user-images.githubusercontent.com/10544674/141240041-077571e9-48a3-47a9-96f4-7282f1a47fdb.png)

If I open it directly from browser, it says Internal Server Error
![image](https://user-images.githubusercontent.com/10544674/141240055-3f64e6f9-ef7e-4db5-aa50-039f07b9ca55.png)

Please give me some clue to investigate.

Thank you!

**Comment by pattersonbl2** [comment, marked as answer]
This points to an issue with the webserver. I would start troubleshooting the Ec2 instance by SSH into it.

**Reply by hungleon2112** [reply]
hi @Pattersonbl2 , thanks for your reply. I tried to ssh into the EC2 instance but I don't know what should I find in here.
Can you give me some more options to try?

**Reply by pattersonbl2** [reply]
You can run a journalctl command to pull some log reports. If you are not familiar with any server admin work I would just redeploy the stack again using the marketplace template to see if it fixes the issues.

**Reply by hungleon2112** [reply]
hi @Pattersonbl2 , thank you for your reply.
May I ask do you mean I should:
1) Fork the server Reticulum 
2) Modify things needed
3) Manually create server on AWS and deploy Reticulum on it
4) Change RETICULUM_SERVER in hubs .env file
5) Re-deploy hubs by npm run deploy

Thank you!

**Reply by tttigo** [reply]
Same issue here. Using custom client.

**Reply by pattersonbl2** [reply]
Backtracking, This is an issue with the public lambda running in the stack called Nearspark.  The lambda fails due to a sharp module error. 

Module did not self-register: '/var/task/node_modules/sharp/build/Release/sharp.node'.

- Remove the "node_modules/sharp" directory, run "npm install" and look for errors
- Consult the installation documentation at https://sharp.pixelplumbing.com/en/stable/install/
- Search for this error at https://github.com/lovell/sharp/issues

**Reply by tanfarming** [reply]
temporary fix:
1. download this zip https://hubs-cloud.s3.us-west-1.amazonaws.com/ftp/nearksparkzip_0.0.2.zip
2. find the thumbnail lambda -- it should be one named something like `<stackName>-NearsparkApplication-ThumbnailFunction-<randomString>`
3. in the lambda's code source section, select "Upload from" > .zip file, and give it nearksparkzip_0.0.2.zip

we'll get with AWS and get the update published in the next few days

**Reply by tttigo** [reply]
worked here, thanks

**Reply by vjFred** [reply]
@tanfarming let us know when on AWS is this resolved, thanks

**Reply by pbchau92dn** [reply]
thanks @tan for quick fixed, upload new template for nearkspark
this worked for me

**Reply by vjFred** [reply]
Hi all, any news on this for those of us who don't have a custom client? Just the AWS template as it is is fixed?

**Reply by pattersonbl2** [reply]
> Hi all, any news on this for those of us who don't have a custom client? Just the AWS template as it is is fixed?

Still waiting to update the CFT. The temp fix does actually work for people without custom clients as well

**Reply by vjFred** [reply]
ok I see, so how can i do this?, is this done before I deploy the AWS template?, i am HC user with little skills in coding. i know how to deploy the stack and stuff but my knowledge on fixing code and replacing is short, i read the fix @tanfarming mention but really can´t understand the steps i have to do to upload the fix into the AWS, so any help will be appreciated, thnks

**Reply by pattersonbl2** [reply]
There is a post in the thread with directions to do this from tanfarming. 

This should work regardless of region.

**Reply by pattersonbl2** [reply]
The marketplace template has been updated. please redeploy stacks using the version 1.1.2.

**Reply by vjFred** [reply]
@Pattersonbl2 Hi Brandon, i have redeploy my personal stack and so far, works good, thnks
![image](https://user-images.githubusercontent.com/69004885/144477057-54e72056-44aa-4bc8-a90e-8e1a0c51854e.png)
